### DISCUSSION TITLE
Match anything after a prefix

### DISCUSSION THREAD

**Discussion by pkese** [Lanayx/Oxpecker#31] [Q&A] [answered]
On a backend data server in development mode I'm serving some static html assets from a production site,
so I'm redirecting all requests starting with /static to some other server.

```fsharp
let redirectHandler targetHost permanent _url : EndpointHandler =
    fun ctx -> redirectTo $"%s{targetHost}{ctx.Request.Path}" permanent ctx

let endpoints = [
  subRoute "/api" [ apiEndpoints ]
  routef "/static/{%s}" (redirectHandler "http://frontend.server" true)
]
```

Problem is that this is redirecting just the files in the root directory but not in subdirectories.

Is there a way to instruct Oxpecker to match everything after the prefix?

I think I could write a custom `Endpoint` and put it into a `subRoute`, but it looks like that requires some boilerplate and doesn't seem to be documented.

**Comment by Lanayx** [comment]
@pkese You should be able to use * or ** like that `route "/static/{**slug}" ... ` (see https://learn.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-8.0#route-templates) and then use `ctx.TryGetRouteValue` to get slug. This technique is not currently available with `routef` and `%s`, but I'll check if this can be added.

**Comment by pkese** [comment]
This works perfectly. Amazing. Thanks.

All it needs is a line or two of documentation.

```fsharp
let redirectHandler targetHost permanent : EndpointHandler =
    fun ctx -> redirectTo $"%s{targetHost}{ctx.Request.Path}" permanent ctx

let endpoints = [
  subRoute "/api" [ apiEndpoints ]
  route "/static/{**path}" (redirectHandler "http://frontend.server" true)
]
```

**Reply by Lanayx** [reply, marked as answer]
I've just published a [new version](https://www.nuget.org/packages/Oxpecker/0.15.0) with catch-all routes support for `routef` and also updated [documentation](https://lanayx.github.io/Oxpecker/src/Oxpecker/#catch-all-route)

**Reply by pkese** [reply]
You really rock.

Oxpecker is such a great library.
